client/my-sites/theme/main.jsx

```diff
@@ -188,10 +192,29 @@ const BannerUpsellTitle = ( {
 	translate,
 	isSiteEligibleForManagedExternalThemes,
 	isMarketplaceThemeSubscribed,
+	isThemeAllowedOnSite,
 } ) => {
 	const bundleSettings = useBundleSettingsByTheme( themeId );
 	const isEnglishLocale = useIsEnglishLocale();
 
+	if ( ! isThemeAllowedOnSite ) {
+		return isEnglishLocale ||
+			i18n.hasTranslation(
+				'Access this theme for FREE with a %(personalPlanName)s, %(premiumPlanName)s, or %(businessPlanName)s plan!'
```

Comment:
When looking at a premium theme this will still say "with a personal, premium or business plan", which isn't true. It also does the same with Woo themes, whereas that doesn't happen in production.

https://github.com/Automattic/wp-calypso/assets/93301/47db9e66-f420-4cc9-8a8b-c4d1ba7dcdb1

Reply:
the links are all correct though, it's just that the messaging needs to be more complicated

Reply:
Nice catch! It should be fixed now.